org.palladiosimulator.analyzer.slingshot.behavior.spd.adjustment/src/org/palladiosimulator/analyzer/slingshot/behavior/spd/monitor/ResourceContainerMonitorCloner.java

```diff
@@ -0,0 +1,167 @@
+	 * 
+	 * @param copiedResourceContainer
+	 * @return
+	 */
+	public List<Monitor> createMonitorsForResourceContainer(final ResourceContainer copiedResourceContainer) {
+		final List<Monitor> monitors = new ArrayList<>(
+				unitResourceContainerMonitors.size() + unitActiveResourceMonitors.size());
+
+		this.unitResourceContainerMonitors.stream()
+				.map(unitMonitor -> createMonitor(copiedResourceContainer, unitMonitor)).forEach(monitors::add);
+
+		this.unitActiveResourceMonitors.stream()
+			.<Monitor>mapMulti((monitor, acceptor) -> {
+				final ActiveResourceMeasuringPoint ap = (ActiveResourceMeasuringPoint) monitor.getMeasuringPoint();
+				copiedResourceContainer.getActiveResourceSpecifications_ResourceContainer().stream()
+						.filter(spec -> isSameProcessingType(spec, ap.getActiveResource()))
+						.findAny()
+						.ifPresent(spec -> acceptor.accept(createMonitor(spec, monitor)));
+			}).forEach(monitors::add);
+
+		return monitors;
+	}
+
+	private static boolean isSameProcessingType(final ProcessingResourceSpecification firstSpec,
+			final ProcessingResourceSpecification secondSpec) {
+		return firstSpec.getActiveResourceType_ActiveResourceSpecification().getId()
+				.equals(secondSpec.getActiveResourceType_ActiveResourceSpecification().getId());
+	}
+
+	private Monitor createMonitor(final ResourceContainer resourceContainer, final Monitor originalMonitor) {
+		final Monitor monitor = createMonitor(originalMonitor);
+		monitor.setMeasuringPoint(createMeasuringPoint(resourceContainer, originalMonitor.getMeasuringPoint()));
+		return monitor;
+	}
+
+	private MeasuringPoint createMeasuringPoint(final ResourceContainer resourceContainer,
+			final MeasuringPoint originalMeasuringPoint) {
+		if (originalMeasuringPoint instanceof ResourceContainerMeasuringPoint) {
+			final ResourceContainerMeasuringPoint copy = PcmmeasuringpointFactory.eINSTANCE
+					.createResourceContainerMeasuringPoint();
+			copy.setMeasuringPointRepository(mpRepository);
+			copy.setResourceContainer(resourceContainer);
+			mpRepository.getMeasuringPoints().add(copy);
+			return copy;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Helper method that creates a generic monitor without a reference to a
+	 * measuring point. Every attribute is copied from the original monitor, except:
+	 * - The reference to the old measuring point - The entity name will contain the
+	 * old name, but with "-copy" appended - The id will be newly generated
+	 * 
+	 * @param originalMonitor The original monitor to copy from
+	 * @return A new monitor
+	 */
+	private Monitor createMonitor(final Monitor originalMonitor) {
+		final Monitor monitor = MonitorRepositoryFactory.eINSTANCE.createMonitor();
+
+		monitor.setActivated(originalMonitor.isActivated());
+		monitor.setEntityName(originalMonitor.getEntityName() + "-copy");
+		monitor.setId(EcoreUtil.generateUUID());
+		monitor.getMeasurementSpecifications().addAll(originalMonitor.getMeasurementSpecifications());
```

Comment:
You cannot just get the MeasurementSpec from a monitor and add them to another one. The relation between Monitors and MeasurementSpecs is _composition_ , i.e. if you add the spec from the original monitor to the new monitor, you thereby remove the spec from the original monitor. 

state of originalMonitor, after creating the first new monitor:
![image](https://github.com/PalladioSimulator/Palladio-Analyzer-Slingshot-Extension-SPD-Interpreter/assets/44230629/d3199013-f501-475e-b1ae-a079ca1e892e)
 
As a follow up, when we create a second new monitor, we put`null` as MeasurementSpecs, because the originalMonitor has noch Specs anymore :(

Reply:
Should be resolved now